eng/cmake/global_compile_options.txt

```diff
@@ -15,5 +15,5 @@ else()
     set(WARNINGS_AS_ERRORS_FLAG "-Werror")
   endif()
 
-  add_compile_options(-Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wcast-qual -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Wfloat-equal)
+  add_compile_options(-Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wcast-qual -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Wfloat-equal -Wno-clobbered)
```

Comment:
How does GCC's `-Wall` compare to MSVC's `/W4`, that @antkmsft recently changed in https://github.com/Azure/azure-sdk-for-c/pull/816? Similarly, what about clang?

cc @barcharcraz, @BillyONeal 

Would we still need the no-clobbered flag if we changed -Wall to -W4?

Reply:
Ah, I see. This is for GCC. We did not change (and probably shouldn't) switch from `-Wall` from GCC: `-Wall` for GCC and CLang is the equivalent of `/W4` for MSVC. `/Wall` for MSVC is an equivalent of `-Weverything` for GCC/Clang.